components/common/Recaptcha.client.vue

```diff
@@ -0,0 +1,103 @@
+<script lang="ts" setup>
+const css = useCssModule();
+
+const props = withDefaults(
+  defineProps<{
+    theme?: 'light' | 'dark';
+    size?: 'compact' | 'normal';
+    invalid?: boolean;
+    useRecaptchaNet?: boolean;
+  }>(),
+  {
+    theme: 'light',
+    size: 'normal',
+    useRecaptchaNet: false,
+  }
+);
+
+const emit = defineEmits<{
+  (e: 'error'): void;
+  (e: 'expired'): void;
+  (e: 'success', value: string): void;
+  (e: 'captcha-id', value: number): void;
+}>();
+
+const {
+  public: {
+    recaptcha: { siteKey },
+  },
+} = useRuntimeConfig();
+
+const recaptchaEl = ref<HTMLElement | null>(null);
+const rendered = ref(false);
+
+// this will be populated once the deferred captcha script is loaded
+const grecaptcha = toRef(window, 'grecaptcha');
+
+/**
+ * Renders the captcha if not rendered and required fields are available
+ */
+const renderCaptcha = () => {
+  if (rendered.value || !(grecaptcha.value && recaptchaEl.value)) {
+    return;
+  }
+
+  const id = grecaptcha.value.render(recaptchaEl.value, {
+    sitekey: siteKey,
+    callback: (token: string) => emit('success', token),
+    'expired-callback': () => emit('expired'),
+    'error-callback': () => emit('error'),
+    theme: props.theme,
+    size: props.size,
+  });
+
+  emit('captcha-id', id);
+  rendered.value = true;
+};
+
+window.onRecaptchaLoaded = renderCaptcha;
+
+onMounted(() => {
+  if (!grecaptcha.value) {
+    useHead({
+      script: [
+        {
+          src: `${
+            props.useRecaptchaNet
+              ? 'https://www.recaptcha.net/recaptcha'
+              : 'https://www.google.com/recaptcha'
```

Comment:
what is the difference between the two domains?

Reply:
The main difference between the two is that https://www.google.com/recaptcha is the recommended and supported version, while https://www.recaptcha.net/recaptcha is an alternative domain that serves the same purpose
@kelsos